python/dgl/transform.py

```diff
@@ -575,6 +575,45 @@ def partition_graph_with_halo(g, node_part, num_hops):
         subg_dict[i] = subg
     return subg_dict
 
+def metis_partition(g, k, num_hops=1):
+    '''
+    This is to partition a graph with Metis partitioning.
+
+    The partitioned graph is stored in DGLGraph. The DGLGraph has the `inner_node` node data
+    that indicates whether a node is inside a partition and has the `inner_edge` edge data
+    that indicates whether an edge is inside a partition. In addition, it has the `part_id`
+    node data that indicates the partition a node belongs to.
```

Comment:
So the `inner_node` can be used for `partition_book`? How to get the `local2global` tensor in this DGLGraph?

Reply:
We should include this mapping in the `dgl.NID` and `dgl.EID` fields.

Reply:
`part_id` can be used to generate `partition_book`. `parent_nid` can be used to generate `local2global`.

Reply:
`dgl.NID` field stores `parent_nid` in a subgraph?